doc/radosgw/dynamicresharding.rst

```diff
@@ -0,0 +1,51 @@
+=================
+RGW Dynamic Resharding
+=================
+
+.. versionadded:: Luminous
+
+The dynamic resharding feature automatically increases the number of bucket index shards.
```

Comment:
Maybe to be more clear we mention dynamic bucket index resharding, though I'm ok either way